cs3342_09-master/README.md

```diff
@@ -0,0 +1,191 @@
+# Grammars Assignment (50 points)
+
+This file contains two questions. The first has 2 parts, the second has 7.
+
+For all but questions 2.3 and 2.5 you'll answer questions by editing this file.
+Your answers will appear in the Ax.y section after each question, replacing the
+placeholders.
+
+For question 2.3, you'll need to add an image file to the repository (.png or
+.pdf) and put its file name as the answer.
+
+For question 2.5, you'll write one or more source code files and add them to the
+repository. You will then show the command used to run the files in the answer.
+Do _not_ upload any binary files for this section.
+
+Please do not edit anything outside the answers sections.
+
+
+# Q1: Mr Fussy Builds a Path
+
+# Q1
+
+Mr Fussy wants to build a path using a row of large square pavers. His path is
+one paver wide and an odd number of pavers long. Pavers come in three colors:
+red, green, blue.
+
+Mr Fussy has a rule: the sequence of colors must be symmetrical across the
+length of the path: the last paver must be the same color as the first, the
+second to last the same color as the second, and so on.
+
+This is a valid path:  rgbgr
+So is this: rrr
+
+This is not: rgb
+This is not: rggr    (it must be an odd length)
+
+## Q1.1  (5 points)
+
+Write a Chomsky type 2 grammar that describes any valid path containing at
+least one paver. Use `S` as the start state, and `r`, `g`, `b` as terminals that
+represent the tiles.
+
+## A1.1
+
+
+S -> color1 | color2 S color2
+
+color -> r|g|b
+
+## Q1.2  (1 point for the O() answer, 2 for the sentence)
+
+Using big-O notation, what is the likely memory requirement for a parser that
+can validate a particular path configuration, where `n` is the number of tiles?
+In one sentence, explain why.
+
+## A1.2
+
+O(1) becuase the parser would only need to iterate over the path one time 
+
+
+# Q2
+
+A simple sentence structure might be
+
+    "the" (zero or more adjectives) noun verb (optional adverb)
+
+Adjectives are "lazy" or "smelly"
+
+Nouns are "dog" or "cat"
+
+Verbs are "ate" and "ran"
+
+Adverbs are "slowly" and "noisily"
+
+The following are examples of valid sentences:
+
+* The smelly dog ran.
+* The smelly dog ran slowly.
+* The cat ate noisily.
+
+## Q2.1 (7 points)
+
+Write the BNF (not EBNF) description for this language.
+
+## A2.1
+
+\<sentence> ::= "the" + \<adj> + \<noun> + \<verb> + \<adverb>
+\<adj> ::= \<<empty>> | "lazy" | "smelly"
+\<adverb> ::=  \<<empty>> | "slowly" | "noisily"
+\<noun> ::=   "dog" | "cat"
+\<verb> ::=   "ate" | "ran"
```

Comment:
I don't know what the backslashes are for and I'm ignoring them

The plus signs are not BNF (-1) and the grammar doesn't allow for multiple adjectives (-2)

# -3